src/pages/signin/LoginForm.js

```diff
@@ -125,6 +125,12 @@ function LoginForm(props) {
             return;
         }
 
+        // Every input that starts with '+' should be validated as a phone number.
+        if (loginTrim.startsWith('+') && !ValidationUtils.isNumericWithSpecialChars(loginTrim)) {
```

Comment:
@akinwale What case can this happen?

Isn't this considered an invalid email?

![Screenshot 2023-08-04 at 20 02 39](https://github.com/Expensify/App/assets/25520267/64427c73-0a1d-4ab6-a770-e317de6f9b76)

Reply:
@mollfpr This can be considered a valid email address, yes, but this is a very rare edge case that is unlikely to happen. Also, based on https://github.com/Expensify/App/issues/23803#issuecomment-1658629913, the input starting with a "+" should have the phone number error displayed (regardless of the characters that come afterwards).

It was also mentioned in previous comments in the issue thread that as long as the input starts with a "+" sign, it will be considered a phone number. See https://github.com/Expensify/App/issues/23803#issuecomment-1658528672 for instance.

Reply:
There is a slim chance that someone might interpret it as a phone number due to the presence of numbers and a plus sign. It does not adhere to a typical phone number format. It is more likely to be an incorrect or invalid email-like string.

I'm afraid that someone will think differently after this PR is merged. 

If the value is not a standard email address that should have the "@" symbol, separating the local part (before "@"), and the domain part (after "@"), with plus sign leading, then we can consider this an invalid phone number. E.g. `+3232asdas323232`.

Reply:
Can you clarify the change I should make here? Here's what I was able to understand:
1. If the input contains an '@' even if it starts with '+', then it should be treated as an email, and show the corresponding error message for invalid email address. I can add another conditional check for this.
2. In every other case where it doesn't look like an email, for example: `+aa228833aa.cc` it will be considered as an invalid phone number, and the phone number invalid message will be shown.

Are these assumptions correct?

Reply:
I think the second case is more appropriate with an email address error because it has a word and `.` for the domain part of an email address.

Reply:
>I think the second case is more appropriate with an email address error because it has a word and . for the domain part of an email address.

I don't think it really matters much, because an email address must always contain an '@' sign.

However, in this case, it looks like the solution would be to only consider inputs that contain digits (numeric 0-9) and the special symbols (`+`, `(`, `(`) and `-`) as a phone number (show phone number error), and then all other cases as an email address (show email address error), in which case, the only thing that needs to be updated is the regex, and I can remove this check: ` if (loginTrim.startsWith('+') && !ValidationUtils.isNumericWithSpecialChars(loginTrim)) {`.

Reply:
Yeah, the updated regex seems enough.

Reply:
Pushed a new commit. All tests run ok as expected.